feat(manager/composer): support git-tags hostRules for github.com when updating artifacts (#18004)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

Any `hostRules` token for `github.com` or `gitlab.com` are found and written out to `COMPOSER_AUTH` in env for Composer to parse.
Any `hostRules` with `hostType=packagist` are also included.

> For dependencies on `github.com` without a packagist server, a hostRule with `hostType=git-tags` should be used with a personal access token (not an application token).

Reviewer: ```suggestion
For dependencies on `github.com` without a Packagist server: use a Personal Access Token for `hostRule` with `hostType=git-tags`, do not use an application token.
```

> Do not add a hostRule with `hostType=github` because it can override the default renovate application token for everything else and cause unwanted side effects.

Reviewer: ```suggestion
Avoid adding a `hostRule` with `hostType=github` because:

- it overrides the default Renovate application token for everything else 
- it causes unwanted side effects.
```

> The repository in `composer.json` should have the `vcs` type with a `https` url. Ex:

Reviewer: ```suggestion
The repository in `composer.json` should have the `vcs` type with a `https` URL.
For example:
```

Any `hostRules` token for `github.com` or `gitlab.com` are found and written out to `COMPOSER_AUTH` in env for Composer to parse.
Any `hostRules` with `hostType=packagist` are also included.
For dependencies on `github.com` without a Packagist server: use a Personal Access Token for `hostRule` with `hostType=git-tags`, do not use an application token.
Avoid adding a `hostRule` with `hostType=github` because:

- it overrides the default Renovate application token for everything else 

> - it causes unwanted side effects.

Reviewer: ```suggestion
- it causes unwanted side effects
```

Whoops, we don't use punctuation at the end of a list item. 🙂
